irohad/ordering/impl/on_demand_ordering_gate.cpp

```diff
@@ -48,14 +50,22 @@ OnDemandOrderingGate::OnDemandOrderingGate(
                   current_round_.block_round, current_round_.reject_round, {});
             }));
       })),
+      cache_(std::move(cache)),
       proposal_factory_(std::move(factory)),
       current_round_(initial_round) {}
 
 void OnDemandOrderingGate::propagateBatch(
-    std::shared_ptr<shared_model::interface::TransactionBatch> batch) const {
+    std::shared_ptr<shared_model::interface::TransactionBatch> batch) {
   std::shared_lock<std::shared_timed_mutex> lock(mutex_);
 
-  network_client_->onBatches(current_round_, {batch});
+  auto batches = cache_->clearFrontAndGet();
```

Comment:
Otherwise, every time new batch is propagated it will be concatenated with the content of the front of the cache and some batches will be sent to network multiple times. It can be reworked when instead of single batch ordering gate receives the list of all batches for current round